src/sentry/utils/arroyo_producer.py

```diff
@@ -38,7 +39,8 @@ def produce(self, destination: Union[Topic, Partition], payload: KafkaPayload) -
     def _get(self) -> KafkaProducer:
         if self._producer is None:
             self._producer = self._factory()
-            atexit.register(self._shutdown)
+
+            register_shutdown(self._shutdown)
```

Comment:
Isn't this going to work only for producers used inside Celery workers? 

If we make this class Celery specific I think we should make it more explicit in the docstring and with the name.